### PR TITLE
Upgrade frontend dependencies

### LINKED ISSUE #1948: Upgrade build process to Webpack 3
## Summary

Currently we use webpack 2 - webpack 3 introduces scope hoisting which should improve the run time of our modules.

### LINKED ISSUE #1981: upgrade major client-side dependencies
We may want to upgrade major dependencies (particularly Vue.js) to get any recent bug and performance improvements. Definitely something to do early - we shouldn't try to do this right before releasing.

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary

Upgrade to webpack3, and upgrade all other frontend dependencies.

## Issues addressed

Fixes #1948 and resolves #1981

### CODE REVIEW

**Comment by rtibbles** [comment]
Weird, this built absolutely fine on my machine. But somehow remotely there are a lot more webpack3 issues. Maybe it never upgraded for real at all...

**Comment by indirectlylit** [comment]
I'm seeing some new issues with Material icons and CORS:

`Access to Font at 'http://localhost:3000/js-dist/MaterialIcons-Regular.woff2?570eb83859dc23dd0eec423a49e147fe' from origin 'http://localhost:8000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:8000' is therefore not allowed access.`

image

File: package.json [hunk 1]
```diff
@@ -32,92 +32,92 @@
...
     "keen-ui": "1.0.0",
     "lodash": "^4.17.4",
     "loglevel": "1.4.1",
-    "markdown-it": "^8.3.1",
+    "markdown-it": "^8.4.0",
     "material-design-icons": "^3.0.1",
-    "purecss": "^0.6.2",
-    "rest": "1.3.2",
+    "purecss": "^1.0.0",
+    "rest": "2.0.0",
```
Comment (indirectlylit) [inline package.json:49]: major version jump - perhaps too risky?

File: package.json [hunk 2]
```diff
@@ -32,92 +32,92 @@
...
     "css-element-queries": "^0.4.0",
-    "date-fns": "^1.28.2",
+    "date-fns": "^1.28.5",
     "engine-strict": "^1.0.0",
-    "filesize": "3.3.0 ",
-    "frame-throttle": "^2.0.1",
-    "intl": "1.2.4",
-    "js-cookie": "2.1.2",
+    "filesize": "3.5.10",
+    "frame-throttle": "^3.0.0",
```
Comment (indirectlylit) [inline package.json:40]: major version jump - perhaps too risky?

**Comment by rtibbles** [comment]
The CORS issue should be dev only, but still not ideal.

**Comment by indirectlylit** [comment]
closing in favor of #2366

**Pull request closed without merging**